synthesizer/program/src/resources/credits.aleo

```diff
@@ -31,6 +31,18 @@ struct committee_state:
     microcredits as u64;
     // The boolean flag indicating if the validator is open to new stakers.
     is_open as boolean;
+    // The percent (0-100) of delegator commissions which validator retains.
+    // Set by validator on self bond.
+    commission as u8;
+
+/**********************************************************************************************************************/
+
+// This mapping tracks total amount bonded to an address.
+mapping delegated:
+    // The key represents the address stake is delegated to. It may not yet be a validator.
+    key as address.public;
```

Comment:
I see a few problems here. I don't see functions to deposit to, withdraw from, or keep track of who the deposits are coming from here. For a single funder, you could maybe assume it's coming from the withdraw key. If multiple funders, you get issues arising of who can force an unbond_delegator_as_validator and unbond_public.

Reply:
The key insight of this approach is that we are only tracking enough state that allows us to adjust the single logical test of if a validator can bond or not. No other logic is adjusted. The risks to evaluate with this approach:

1. Limit the data size of `delegate` (which is done with higher bond requirement).
2. Does this approach still provide the same limits around `committee` size as are currently present.

All of that state of where funds are coming from is already tracked in the `bonded` mapping. We take advantage of that to limit the complexity of what this needs to provide. Modifications to `finalize unbond_delegator_as_validator()` show this working.

For a bonded validator the value of `delegated[r0]` is the same as `committee[r0].microcredits`. It is redundant. 

However that redundancy is beneficial as it does not impact any existing logic and tradeoffs that have already been weighed out for the interaction between `bonded` and `committee`.

Additionally the reason for adding a new `delegated` instead of modifying how `committee` works is that we are sure to not impact any snarkVM logic. Concivably this same approach could have been implementd by adding a flag to `struct committee_state`, but then it would require snarkVM changes.

I will flush out the changes to `finalize unbond_public()` so you can see, but they will be of the same substance as `finalize unbond_delegator_as_validator()`. Both of those functions will also need an additional test to remove a validator from the committee if stake drops below a specified threshold. This would allow either a delegator or the validator itself to remove a validator from the `committee`. If the validator has self bonded sufficient credits, then it does not matter what delegators do.

As a possible cleanup with this idea (but was more of a change set than I was prepared to make in a couple hours) is that the need to call `unbond_delegator_as_validator()` is really not necessary.